azurelinuxagent/ga/monitor.py

```diff
@@ -287,6 +288,19 @@ def add_sysinfo(self, event):
                 event.parameters.remove(param)
         event.parameters.extend(self.sysinfo)
 
+    def update_container_id(self, event):
+        last_known_container_id = self.protocol.client.get_container_id_from_goal_state()
```

Comment:
can self.protocol be an instance of Metadata? (Azure Stack)

Reply:
It can, in which case any reference to `self.protocol.client` would break.
Except this instance, we already have another one checked in :( Here it is: https://github.com/Azure/WALinuxAgent/blame/develop/azurelinuxagent/ga/monitor.py#L336

Reply:
Removed.